frontend/src/tests/application-tests.js

```diff
@@ -1,25 +1,268 @@
+
+        // get the current active user
+        let resp = await apiGET("/debug/active_user");
+        expect(resp).toHaveStatus("success");
+        defaultUser = resp.payload;
     }, 30000);
 
     // These tests set data through the `/public/postings` route,
     // but read data through the `/api/v1/admin` route.
     describe("Public route tests", () => {
-        it.todo("Get survey.js posting data through public route");
+        it("Get survey.js posting data through public route", async () => {
+            // make sure the user has admin permissions before this post request
+            // make a new posting and update <posting> to include the id of the posting
+            resp = await apiPOST("/admin/postings", {
+                ...posting,
+                session_id: session.id,
+            });
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(postingPropTypes, resp.payload);
+            expect(resp.payload.id).not.toBeNull();
+            Object.assign(posting, resp.payload);
+
+            // make sure the user is the seeded applicant before this get request for the purposes of the next test
+            // create a user for the applicant to make sure they exist before changing the default user to applicant
+            const applicantAsUser = {
+                roles: ["ta"],
+            };
+            applicantAsUser.utorid = applicant.utorid;
+
+            resp = await apiPOST("/debug/users", applicantAsUser);
+            expect(resp).toHaveStatus("success");
+            Object.assign(applicantAsUser, resp.payload);
+
+            // set the default user to the applicant
+            resp = await apiPOST("/debug/active_user", applicantAsUser);
+            expect(resp).toHaveStatus("success");
+
+            // read survey.js posting data
+            resp = await apiGET(`/public/postings/${posting.url_token}`, true);
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            surveyPrefill = resp.payload["prefilled_data"];
+
+            // restore the default user
+            resp = await apiPOST("/debug/active_user", defaultUser);
+            expect(resp).toHaveStatus("success");
+        });
+
+        // broken up into 2 tests: applicant & application(not yet implemented in the backend)
+        it("Survey.js posting data is pre-filled based on prior applicant", async () => {
+            // this test depends on the previous test on the terms of surveyPrefill variable
+            const prefilled = [
+                "utorid",
+                "student_number",
+                "first_name",
+                "last_name",
+                "email",
+                "phone",
+            ];
+            for (let prefilledKey of prefilled) {
+                // check if every applicant information is in the prefilled data
+                expect(surveyPrefill[prefilledKey]).toBeDefined();
```

Comment:
You check if it's the right thing immediately afterwards. You don't need to check if it's defined.